### PR TITLE
add update_nested

### PR DESCRIPTION
**Pull request by FynnBe**
as discussed in https://github.com/bioimage-io/core-bioimage-io-python/pull/256

### CODE REVIEW

**Review by constantinpape — COMMENTED** [review]
The changes look good, but this is missing some user-facing documentation (and functionality?):
How can we use this to directly update a rdf.yaml file? 
Do we need additional functionality for this? If yes, should it go into core?

**Comment by FynnBe** [comment]
Yes, let's maybe add a cli command `update-rdf` with the (default?) option `--validate` that takes RDF, UPDATE, OUTPUT as arguments. 
The other use cases I see for now are:
 - updates in collection (internal)
 - updates in `export_bioimageio_model` (internal)

**Comment by constantinpape** [comment]
> Yes, let's maybe add a cli command `update-rdf` with the (default?) option `--validate` that takes RDF, UPDATE, OUTPUT as arguments.

That sounds good, but `update_rdf` should also be available as a proper function in `bioimageio.spec`, so that it can be used programatically and not only through the CLI.

**Comment by FynnBe** [comment]
> > Yes, let's maybe add a cli command `update-rdf` with the (default?) option `--validate` that takes RDF, UPDATE, OUTPUT as arguments.
> 
> That sounds good, but `update_rdf` should also be available as a proper function in `bioimageio.spec`, so that it can be used programatically and not only through the CLI.

how about
```python
def update_rdf(
    source: Union[dict, os.PathLike, IO, str, bytes],
    update: Union[dict, os.PathLike, IO, str, bytes],
    output: Union[dict, os.PathLike] = None,
    validate: bool = True,
) -> Union[dict, Path]:
   ...
```
the default would be to return a dict

**Comment by constantinpape** [comment]
> how about
> 
> ```python
> def update_rdf(
>     source: Union[dict, os.PathLike, IO, str, bytes],
>     update: Union[dict, os.PathLike, IO, str, bytes],
>     output: Union[dict, os.PathLike] = None,
>     validate: bool = True,
> ) -> Union[dict, Path]:
>    ...
> ```
> 
> the default would be to return a dict

Yeah, that's good!
Can you implement s.t. "inplace" updates are possible? I.e. if `ouptut=source` the input rdf file should be updated directly.

File: bioimageio/spec/commands.py [hunk 1]
```diff
@@ -120,3 +160,67 @@ def validate(
...
+        output:  dict or path to write output to (default: return new dict)
+        validate_output: whether or not to validate the updated RDF
+
+    Returns:
+        The updated content of the source rdf as dict or,
+        if output is a path, that path (where the updated content is saved to).
+
+    Raises:
+        ValidationError: if `validate_output` and the updated rdf does not pass validation
+    """
```
Comment (FynnBe) [inline bioimageio/spec/commands.py:184]: very close to the initial draft, I only added `RawResourceDescription` for source and update, so one does not have to go back to dicts if the RDF(s) are already loaded.

**Review by constantinpape — APPROVED** [review]
LGTM

**Pull request merged**